src/components/browse/components/FacetList/index.js

```diff
@@ -285,7 +349,52 @@ export class FacetList extends React.PureComponent {
                 }))
             );
             return <Facet {...commonProps} facet={facet} key={facet.field} {...{ defaultFacetOpen, isStatic }} />;
+        }
+
+        const allFacets = []; // first populated with ungrouped assets, then grouped assets are spliced in
+        const groupedOnly = []; // only facet groups
+        const inGroupIndices = {}; // map group names to index in GroupedOnly for quick lookup
+        const spliceIndices = {}; // map group names to index in allFacets where a nested facet should be spliced
+
+        useFacets.forEach(function(facet, i){
+            if (!facet.grouping) {
+                allFacets.push(generateFacet(facet, i-1)); // add ungrouped facets straight to allFacets
+            } else { // add or update facet groups and store in groupedOnly
+                const terms = this.memoized.mergeTerms(facet, filters);
+                const areTermsSelected = this.memoized.anyTermsSelected(terms, facet, filters);
+
+                // keep track of what index in allFacets a particular group should be added at
+                if (!spliceIndices.hasOwnProperty(facet.grouping)) {
+                    spliceIndices[facet.grouping] = allFacets.length;
+                }
+
+                // check if there's a facet group in groupedOnly;
+                if (inGroupIndices.hasOwnProperty(facet.grouping)) {
+                    const i = inGroupIndices[facet.grouping];
+                    groupedOnly[i].facets.push(generateFacet(facet, i));
+
+                    // if any terms are selected, update group selected status
+                    if (!groupedOnly[i].areTermsSelected && areTermsSelected) {
+                        groupedOnly[i].areTermsSelected = true;
+                    }
+                } else  {
+                    groupedOnly.push({
+                        key: facet.grouping,
+                        title: facet.grouping,
+                        facets: [generateFacet(facet, i)],
+                        areTermsSelected
+                    });
+                    inGroupIndices[facet.grouping] = groupedOnly.length - 1;
+                }
+            }
+        }.bind(this));
```

Comment:
Prefer to use arrow func (`()=> { }`) if need to bind.
(Subjective on my part probably:) For anonymous functions especially; makes it easier when reading through code to understand where pure function vs a method or bound function is by presence of the `function` keyword, but wouldn't really work if also using `bind`.

Reply:
Ah okay. I think it's just force of habit from writing ES5 at my old job. I'll try to remember to use those.